.github/workflows/emscripten.yml

```diff
@@ -0,0 +1,88 @@
+# Copied from NumPy https://github.com/numpy/numpy/pull/25894
+# https://github.com/numpy/numpy/blob/d2d2c25fa81b47810f5cbd85ea6485eb3a3ffec3/.github/workflows/emscripten.yml
+#
+
+name: Test Emscripten/Pyodide build
+
+on:
+  pull_request:
+    branches:
+      - main
+      - maintenance/**
+  # TODO: refine after this is ready to merge
+  push:
+    # branches:
+    #   - main
+    #   - maintenance/**
+  workflow_dispatch:
+
+env:
+  FORCE_COLOR: 3
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
+
+jobs:
+  build-wasm-emscripten:
```

Comment:
> Right now this is only building and testing and not uploading the wheel somewhere, correct?
>
> It's fine if we keep that out of scope for this PR but I'd still like to talk about how we might include the wheel in our [doc building workflow](https://github.com/scikit-image/scikit-image/blob/7ae4fd8833391a3fa6f28b12a53a177699518764/.github/workflows/build_docs.yml)?

Yes, that is correct. The doc builds workflow(s) aspect is still something I am experimenting with right now, but it has been completed for SciPy (https://github.com/scipy/scipy/pull/20019) and PyWavelets (https://github.com/PyWavelets/pywt/pull/728). The SciPy one is disabled right now, but the PyWavelets one is available and can be accessed on the PyWavelets documentation website at any of the examples under https://pywavelets.readthedocs.io/en/latest/ref/index.html. The issue we have at hand here is how to provide these wheels pre-installed into a JupyterLite-compatible kernel through either a custom Pyodide distribution or through any other mechanism, which I am actively working towards.

> We might also upload it as part of [nightly wheel building workflow](https://github.com/scikit-image/scikit-image/blob/main/.github/workflows/nightly_wheel_build.yml?rgh-link-date=2024-04-13T11%3A43%3A07Z) similar to [numpy](https://github.com/numpy/numpy/blob/ab7649fe2ed8f0f0260322d66631b8dfab57deff/.github/workflows/emscripten.yml#L116-L126).
> 
> In case we decide tackle this, may we ping you for support? :D

Absolutely! I would be more than happy to make the changes myself in a follow-up PR, considering that this would be beneficial for the documentation builds mentioned above – the plan will be to use these wheels, after they have been built, but only in the latest versions of the docs (the stable versions may use a pinned wheel from the Pyodide CDN), however, this is preliminary and I suspect things will change. I hope I can trouble you for a review on another PR soon after this one ;)

Reply:
Perfect. Then let's get to that after merging this PR. :D

Reply:
Ideally, we'd store the built wheel as an artifact, grab it from the docs build, and upload it as part of the docs. @agriyakhetarpal it should be possible to mpip or similar from there?

Reply:
Yes. This is something that I am trying to get done for PyWavelets first through the [provided artifacts](https://jupyterlite.readthedocs.io/en/latest/howto/pyodide/wheels.html), but this is still in-progress work. For installing from `pip`, JupyterLite provides [abstractions over `micropip`](https://jupyterlite.readthedocs.io/en/latest/howto/pyodide/packages.html#installing-packages-at-runtime) to install packages (it won't work for Anaconda.org packages of the [lack of CORS headers](https://github.com/pyodide/micropip/issues/101) and a host of other reasons, but the Anaconda index at least serves as a nice place to host these wheels going forward). I would suggest looking at the Anaconda upload and then serving them through the docs in separate PRs so that they are easier to review for the maintainers.